crates/ark/src/modules/positron/system.R

```diff
@@ -21,3 +21,11 @@ has_x11 <- function() {
     out <- as.list(vapply(cats, Sys.getlocale, "string", USE.NAMES = TRUE))
     c(LANG = Sys.getenv("LANG"), out)
 }
+
+#' Reports a vector of environment variables for the R process.
+#' @returns Named character vector of env vars that start with the given regex.
+#' @export
+.ps.rpc.get_env_vars <- function(pattern) {
+    matches <- grep(pattern, names(Sys.getenv()), value = TRUE)
```

Comment:
Will the return value be too unstable to safely work with, since the grep pattern might match unexpected variables? Alternatively the string could be split by a separator like `;`, and each of the components retrieved in order. This way there is an invariant relationship between inputs and outputs.

Reply:
Can you give me an example of the kind of unexpected variable you are thinking of?

Reply:
I would much prefer explicit extraction of env vars rather than a regex approach

Reply:
So for the [motivating testthat example](https://github.com/posit-dev/positron/issues/2723), you would want to pass over every `TESTTHAT*` env var explicitly (and need to update the list if testthat adds one) rather than choose them via their prefix?

Reply:
On second thought, since the goal is to reproduce the environment for specific packages, matching a prefix seems like a reasonable way of doing that.

But I wonder if these variables should be pushed from ark to the frontend via the UI comm after each top-level execution to ensure the frontend is up-to-date without getting into concurrency issues of a pull approach (will R be busy when an update is needed, etc).

Reply:
The motivating example here is to use these environment variables in _tasks_ so I think what we have here is a good option, given those constraints. We have to pass the env var into the task object, similar to how we get the pandoc path here: 
https://github.com/posit-dev/positron/blob/348f25f96745f00beed473b6d000bf6b7e6889c8/extensions/positron-r/src/tasks.ts#L65

These env vars need to get passed through via the session manager, much like the locale info that gets used here:
https://github.com/posit-dev/positron/blob/348f25f96745f00beed473b6d000bf6b7e6889c8/extensions/positron-r/src/session.ts#L358